container/container_unix.go

```diff
@@ -280,18 +281,50 @@ func (container *Container) UnmountSecrets() error {
 
 // UpdateContainer updates configuration of a container.
 func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfig) error {
+	sysInfo := sysinfo.New(true)
+
 	container.Lock()
 	defer container.Unlock()
 
 	// update resources of container
 	resources := hostConfig.Resources
 	cResources := &container.HostConfig.Resources
+
+	// validate NanoCPUs, CPUPeriod, and CPUQuota
+	if resources.NanoCPUs > 0 && resources.CPUPeriod > 0 {
```

Comment:
Given that some of these validations depend on what _was_ set in the container, perhaps it's needed to;

1. Perform validation of the options that were set (`daemon.verifyContainerSettings()`)?
2. Merge new configuration with existing configuration of the container (this still needs to validate options that are not allowed to be _updated_ - although there already seems to be an `update` boolean [to use different validation steps](https://github.com/docker/docker/blob/f7819fcb25aaf5b08202db275847c825e7b09999/daemon/daemon_unix.go#L315-L317))
3. Perform the validation again

I haven't checked all scenarios, so hope it makes sense; basically trying to keep validation central where possible